src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -82,71 +85,36 @@ public static Checksum Create(ReadOnlySpan<Checksum> hashes)
     }
 
     public static Checksum Create(ArrayBuilder<Checksum> checksums)
-    {
-        using var stream = SerializableBytes.CreateWritableStream();
-
-        using (var writer = new ObjectWriter(stream, leaveOpen: true))
+        => Create(checksums, static (checksums, writer) =>
         {
             foreach (var checksum in checksums)
                 checksum.WriteTo(writer);
-        }
-
-        stream.Position = 0;
-        return Create(stream);
-    }
+        });
 
     public static Checksum Create(ImmutableArray<Checksum> checksums)
-    {
-        using var stream = SerializableBytes.CreateWritableStream();
-
-        using (var writer = new ObjectWriter(stream, leaveOpen: true))
+        => Create(checksums, static (checksums, writer) =>
         {
             foreach (var checksum in checksums)
                 checksum.WriteTo(writer);
-        }
-
-        stream.Position = 0;
-        return Create(stream);
-    }
+        });
 
     public static Checksum Create(ImmutableArray<byte> bytes)
-    {
-        using var stream = SerializableBytes.CreateWritableStream();
-
-        using (var writer = new ObjectWriter(stream, leaveOpen: true))
+        => Create(bytes, static (bytes, writer) =>
         {
-            for (var i = 0; i < bytes.Length; i++)
-                writer.WriteByte(bytes[i]);
-        }
-
-        stream.Position = 0;
-        return Create(stream);
-    }
+            foreach (var b in bytes)
+                writer.WriteByte(b);
+        });
 
     public static Checksum Create<T>(T value, ISerializerService serializer)
     {
-        using var stream = SerializableBytes.CreateWritableStream();
         using var context = new SolutionReplicationContext();
 
-        using (var objectWriter = new ObjectWriter(stream, leaveOpen: true))
-        {
-            serializer.Serialize(value!, objectWriter, context, CancellationToken.None);
-        }
-
-        stream.Position = 0;
-        return Create(stream);
-    }
-
-    public static Checksum Create(ParseOptions value, ISerializerService serializer)
```

Comment:
this helper went away entirely.  the one place that was calling it just inlined its behavior at the callsite.